### PR TITLE
fix: preserve URL encoding in lowercaseUrlOriginAndRemoveTrailingSlash

### PR DESCRIPTION
**Pull request by Piyush0049**
Closes #18698

## Context
My earlier PR for this issue was closed as it didn't meet quality standards — I've addressed that feedback and am resubmitting a cleaner version with all lint, formatting, and edge cases resolved.

## Problem
[lowercaseUrlOriginAndRemoveTrailingSlash](cci:1://file:///d:/rocket_chat/twenty/packages/twenty-shared/src/utils/url/lowercaseUrlOriginAndRemoveTrailingSlash.ts:3:0-36:2) uses [safeDecodeURIComponent()](cci:1://file:///d:/rocket_chat/twenty/packages/twenty-shared/src/utils/url/safeDecodeURIComponent.ts:0:0-6:2) on `url.pathname` and `url.search`, which decodes percent-encoded characters like `%2F` → `/`. This breaks Google Maps links and other URLs where encoding is significant.

**Example:**
- ✅ Original: `...!16s%2Fg%2F1ptwh8096...`
- ❌ After save: `...!16s/g/1ptwh8096...` (broken link)

## Fix
- Extract path/query/hash directly from the raw URL string instead of using the URL API's parsed (and auto-decoded) components
- Guard against non-HTTP(S) protocols (e.g., `file:`, `data:`) where `url.origin` returns `'null'`
- Remove the unused [safeDecodeURIComponent](cci:1://file:///d:/rocket_chat/twenty/packages/twenty-shared/src/utils/url/safeDecodeURIComponent.ts:0:0-6:2) import

## Testing
- All 14 unit tests pass
- Lint and Prettier formatting pass
- Updated test expectations to verify encoding is preserved, not decoded

### CODE REVIEW

File: packages/twenty-shared/src/utils/url/lowercaseUrlOriginAndRemoveTrailingSlash.ts [hunk 1]
```diff
@@ -9,11 +8,43 @@ export const lowercaseUrlOriginAndRemoveTrailingSlash = (rawUrl: string) => {
...
+  const queryStartIndex = rawUrl.indexOf('?', protocolIndex + 3);
+  const hashStartIndex = rawUrl.indexOf('#', protocolIndex + 3);
+
+  if (pathStartIndex !== -1) endOfHost = Math.min(endOfHost, pathStartIndex);
+  if (queryStartIndex !== -1) endOfHost = Math.min(endOfHost, queryStartIndex);
+  if (hashStartIndex !== -1) endOfHost = Math.min(endOfHost, hashStartIndex);
+
+  const rawSuffix = rawUrl.substring(endOfHost);
+
+  return (lowercaseOrigin + rawSuffix).replace(/\/$/, '');
```
[the remark this answers is not in the document]
Comment (Piyush0049) [reply packages/twenty-shared/src/utils/url/lowercaseUrlOriginAndRemoveTrailingSlash.ts:49]: This was the original approach that caused the bug. The URL API normalizes/re-encodes components — e.g., `ménard` in the raw URL becomes `m%C3%A9nard` via `url.pathname`, silently changing the stored URL. Our raw-string extraction preserves the original encoding exactly as provided.

**Comment by Piyush0049** [comment]
This PR is ready for review. All CI checks, lint, and 19 unit tests pass. I've addressed every automated review comment and added edge-case coverage for opaque origins, blob: URLs, and backslash paths. Would appreciate a review when you get a chance. Happy to make any adjustments. Thanks!

File: packages/twenty-shared/src/utils/url/lowercaseUrlOriginAndRemoveTrailingSlash.ts [hunk 2]
```diff
@@ -9,11 +8,43 @@ export const lowercaseUrlOriginAndRemoveTrailingSlash = (rawUrl: string) => {
...
-  const path =
-    safeDecodeURIComponent(url.pathname) +
-    safeDecodeURIComponent(url.search) +
-    url.hash;
 
-  return (lowercaseOrigin + path).replace(/\/$/, '');
+  const protocolIndex = rawUrl.indexOf('://');
+  if (protocolIndex === -1) {
+    return rawUrl;
+  }
```
[the remark this answers is not in the document]
Comment (Piyush0049) [reply packages/twenty-shared/src/utils/url/lowercaseUrlOriginAndRemoveTrailingSlash.ts:29]: The protocolIndex === -1 fallback intentionally returns rawUrl unmodified as the safest option. Using url.pathname/search/hash would re-encode non-ASCII characters, changing the stored URL.

**Comment by charlesBochet** [comment]
@Piyush0049 Thanks for opening a PR, we have decided to go with a more broader fix: https://github.com/twentyhq/twenty/pull/18792

The help is much appreciated though
